core/base_service.go

```diff
@@ -826,6 +834,16 @@ var (
 	// scheme specified in the URL is invalid. This error isn't typed
 	// specifically so we resort to matching on the error string.
 	schemeErrorRe = regexp.MustCompile(`unsupported protocol scheme`)
+
+	// A regular expression to match the error returned by net/http when a
+	// request header or value is invalid. This error isn't typed
+	// specifically so we resort to matching on the error string.
+	invalidHeaderErrorRe = regexp.MustCompile(`invalid header`)
+
+	// A regular expression to match the error returned by net/http when the
+	// TLS certificate is not trusted. This error isn't typed
+	// specifically so we resort to matching on the error string.
+	notTrustedErrorRe = regexp.MustCompile(`certificate is not trusted`)
```

Comment:
You've probably made a more thorough investigation than me - just right now, but according to the following code snippet from Go's source code, we might need to add a Windows specific regex pattern to make sure this package is truly cross-platform.
```go
macosErr:   "x509: “valid.testing.golang.invalid” certificate is not trusted",
windowsErr: "x509: certificate signed by unknown authority",
```
[source](https://github.com/golang/go/blob/aeac0b6cbfb42bc9c9301913a191bb09454d316a/src/crypto/x509/platform_test.go#L171)

Reply:
When I originally wrote the `IBMCloudSDKRetryPolicy()` function, I essentially stole most of the code from the `go-retryablehttp.ErrorPropagatedRetryPolicy`, located [here](https://github.com/hashicorp/go-retryablehttp/blob/40b0cad1633fd521cee5884724fcf03d039aaf3f/client.go#L486).

It looks like you referenced a section of a testcase.  Are we sure that on windows we do in fact get a different error message?

Reply:
I resolve this thread as we discussed it on Slack. Phil suggested to use
```go
notTrustedErrorRe = regexp.MustCompile(`certificate is not trusted|certificate is signed by unknown authority`)
```
and I think that's a good idea. It should cover most if not all situations.

Reply:
I updated the regex to include both substrings.